src/west/commands/configuration.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2019, Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''West config commands'''
+
+import argparse
+import configparser
+import sys
+from west import config
+from west.config import ConfigFile
+from west.commands import WestCommand
+
+CONFIG_DESCRIPTION = '''\
+West configuration file handling.
+
+Command to assist user in getting or setting west project setting
+'.west/config' or west user configuration file '~/.westconfig'.
+
+To get a value for <name>, type
+    west config <name>
+
+To set a value for <name>, type
+    west config <name> <value>
+'''
+
+
+class Configuration(WestCommand):
+    def __init__(self):
+        super().__init__(
+            'config',
+            'get or set configuration settings in west config files',
+            CONFIG_DESCRIPTION)
+
+    def do_add_parser(self, parser_adder):
+        parser = parser_adder.add_parser(
+            self.name,
+            help=self.help,
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            description=self.description)
+
+        parser.add_argument(
+            'name',
+            help='''name is section and key separated by a dot. As example
+            'west config section.key [value]' to get or set key from the
+            section''')
+
+        parser.add_argument(
+            'value',
+            nargs='?',
+            help='Value to set in config file for the given name')
+
+        mx_group = parser.add_mutually_exclusive_group()
+        mx_group.add_argument('--global',
+                              dest='conf_global',
+                              action='store_true',
+                              help='Use global ~/.westconfig for values')
+        mx_group.add_argument('--local',
+                              action='store_true',
+                              help='''Use project specific .west/config for
+                              values''')
+        mx_group.add_argument('--system',
+                              action='store_true',
+                              help='''Use system specific west config for
+                              values''')
+        return parser
+
+    def do_run(self, args, user_args):
+        configfile = ConfigFile.ALL
+        config_settings = configparser.ConfigParser()
+        if args.conf_global:
+            configfile = ConfigFile.GLOBAL
+        elif args.local:
+            configfile = ConfigFile.LOCAL
+        elif args.system:
+            configfile = ConfigFile.SYSTEM
+
+        name_list = args.name.split(".")
+
+        if len(name_list) < 2:
+            print('west config: error: missing key, please invoke as: west '
+                  'config <section>.<key>')
+            sys.exit(3)
```

Comment:
Prefer not.
In this case i prefer to have the print message look similar to what happens when invoking as `west config` without `<name>` where arg parser handles the error:
```
$ west config
usage: west config [-h] [--global | --local | --system] name [value]
west config: error: the following arguments are required: name
```
so that if only name is given, you'll see this:
```
$ west config name
west config: error: missing key, please invoke as: west config <section>.<key>
```

Reply:
> Prefer not.

`print()` within west is basically never the right thing to do; we use `log` for all output. It allows global changes to how output is organized. This for example won't be colored red which is the expected behavior for west errors.

For example, if we ever add a `--quiet` option, we can do that by having all the `log` routines that print to stderr just return, whereas if we call `print()` we'll have to go throughout the code base and add if statements.

Reply:
agree, should change to but then how about using `log.inf`, instead of `log.die` ?

Reply:
Changed to `log.inf`, let me know if you insist on `log.die`

Reply:
It should be colored red as an error, which means it has to be `log.err` at least. But then since you want to exit immediately it might as well be `log.die`, no?

Reply:
that was my point, arg parser doesn't color this kind of output red, and I consider this an argument error, thus I find it more consistent to align with arg parser, than aligning with other errors which might occur during execution of `west`

I see this error handling similar to what we do here:
https://github.com/zephyrproject-rtos/west/blob/master/src/west/_bootstrap/main.py#L222

But it is not something I will insist strongly upon, so I have changed to `log.die`.